### PR TITLE
Feature/websocket

### PR DESCRIPTION
**Pull request by Dhruvi-Makwana**

### CODE REVIEW

File: Chatify/chat/static/js/register.js [hunk 1]
```diff
@@ -2,7 +2,15 @@ function register(e) {
         e.preventDefault();
         var csrfToken = $('input[name="csrfmiddlewaretoken"]').val();
         var formdata = new FormData($('#register_form')[0]);
-        makeAjaxRequest('POST', csrfToken, "/register/", formdata, function(response) {
+        makeAjaxRequest('POST', csrfToken, "/api/register/", formdata, function (response) {
+             console.log(response)
```
Comment (Dhruvi-Makwana) [inline Chatify/chat/static/js/register.js]: please remove console.log !!

File: Chatify/chat/api.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from rest_framework import status
 from .serializers import UserSerializer, LoginSerializer, GetUserDataSerializer
 from .models import User
+from .utils import *
```
Comment (Dhruvi-Makwana) [inline Chatify/chat/api.py]: don't use * use proper name

File: Chatify/chat/api.py [hunk 3]
```diff
@@ -26,8 +36,48 @@ def get(self, request):
...
 
 
+class LoginAPIView(APIView):
+    def get(self, request):
+        return Response(status=status.HTTP_200_OK)
+
+    def post(self, request, *args, **kwargs):
+
+        login_serializer = LoginSerializer(data=request.data)
+        login_serializer.is_valid()
```
Comment (yatishTT) [inline Chatify/chat/api.py]: is_valid(raise_exception=True)

use try except

File: Chatify/chat/api.py [hunk 4]
```diff
@@ -26,8 +36,48 @@ def get(self, request):
...
+        user = authenticate(request=request, **login_serializer.validated_data)
+        if user:
+            login(request, user)
+            return Response(
+                {"logins": login_serializer.data}, status=status.HTTP_200_OK
+            )
+        else:
+            return Response(
+                login_serializer.errors, status=status.HTTP_406_NOT_ACCEPTABLE
```
Comment (yatishTT) [inline Chatify/chat/api.py]: 400 bad request

File: Chatify/chat/api.py [hunk 5]
```diff
@@ -26,8 +36,48 @@ def get(self, request):
...
+            )
+
+
 class UserOnlineAPIView(APIView):
+    def get(self, request, *args, **kwargs):
+        return Response(status=status.HTTP_200_OK)
```
Comment (yatishTT) [inline Chatify/chat/api.py]: don't need for get method

File: Chatify/chat/api.py [hunk 6]
```diff
@@ -26,8 +36,48 @@ def get(self, request):
...
+
     def post(self, request, *args, **kwargs):
-        user = request.user.id
-        user.save(is_online=True if request.data.get("online") else False)
+
+        data = request.data.copy()
```
Comment (yatishTT) [inline Chatify/chat/api.py]: no need of copy

File: Chatify/chat/api.py [hunk 7]
```diff
@@ -26,8 +36,48 @@ def get(self, request):
...
+        user = request.user
+        user.is_online = (
```
Comment (yatishTT) [inline Chatify/chat/api.py]: user.is_online = data.get("online") is not None or False

File: Chatify/chat/api.py [hunk 8]
```diff
@@ -26,8 +36,48 @@ def get(self, request):
...
+        )
+        user.save()
         return Response(status=status.HTTP_200_OK)
+
+
+class ShowUserData(APIView):
+    def get(self, request):
+        data = User.objects.all()
+        return JsonResponse(
+            {"UserData": list(GetUserDataSerializer(data, many=True).data)}
```
Comment (yatishTT) [inline Chatify/chat/api.py:84]: UserDataSerializer

File: Chatify/chat/consumers.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
+from channels.consumer import SyncConsumer
+from channels.exceptions import StopConsumer
+from asgiref.sync import async_to_sync
+
+
+class MyChatConsumer(SyncConsumer):
```
Comment (yatishTT) [inline Chatify/chat/consumers.py]: What is myconsumer ?

File: Chatify/chat/static/js/chat.js [hunk 10]
```diff
@@ -14,46 +14,20 @@ app.controller('chatCtrl', function($scope, $http) {
...
-                    id: 2
-                }],
-                receiver: [{
-                    user: "karan",
-                    profile: "http://127.0.0.1:8000/static/images/social_media.jpg",
-                    message: "hiiii karan this is receiver karan",
-                    id: 1
-                }]
+    $scope.ajaxGet = function(url, callback = null) {
+        $http.get(url).then(function(response) {
```
Comment (yatishTT) [inline Chatify/chat/static/js/chat.js:27]: use our ajax funciton

**Review by yatishTT — CHANGES_REQUESTED** [review]

File: Chatify/chat/static/js/chat.js (hunk 10 above)
Comment (Dhruvi-Makwana) [reply Chatify/chat/static/js/chat.js:27]: in this js we using angular JS so it's better to create here ajaxGet.

File: Chatify/chat/consumers.py [hunk 11]
```diff
@@ -4,28 +4,12 @@
...
-        #         'message' :event['text']
-        #     })
-
-    # def chat_message(self ,event):
-    #     print("message in handler" ,event)
-    #     self.send({
-    #         'type': 'websocket.send',
-    #         'text': event['message']
-    #     })
+        print("message Recevied", event)
```
Comment (Dhruvi-Makwana) [inline Chatify/chat/consumers.py]: remove print statement

File: Chatify/chat/static/js/register.js [hunk 12]
```diff
@@ -3,14 +3,21 @@ function register(e) {
...
-        
-        })
+            var error = JSON.parse(response.responseText).errors;
+            var username_err = JSON.parse(response.responseText).errors.username;
+            var number = JSON.parse(response.responseText).errors.mobile_number;
+            var pass_error = JSON.parse(response.responseText).errors.password;
+            var mail_error = JSON.parse(response.responseText).errors.email;
+            if(error){
+                    $("#username").html(username_err);
+                    $("#phonenumber").html(number);
```
Comment (Dhruvi-Makwana) [inline Chatify/chat/static/js/register.js:13]: Is it  working properly ?

File: Chatify/chat/templates/chat/register.html [hunk 13]
```diff
@@ -29,19 +29,20 @@
...
       </div>
       <div class="field">
-         <input type="file" class="pt-2" name="profile_photo" required>
+         <input type="file" class="pt-2" name="profile_photo">
       </div>
       <div class="field">
          <input type="submit" value="Register" id="submit" onclick="register(event)">
       </div>
-      <div class="signup-link">Already a user ?..Login</div>
+      <div class="signup-link">Already a user ?..<a href="{% url 'chat:loginUI' %}">Login</a></div>
```
Comment (Dhruvi-Makwana) [inline Chatify/chat/templates/chat/register.html:44]: is it require to write a app name

File: Chatify/chat/static/js/register.js (hunk 12 above)
Comment (Dhruvi-Makwana) [reply Chatify/chat/static/js/register.js:13]: yes

File: Chatify/chat/templates/chat/register.html (hunk 13 above)
Comment (Dhruvi-Makwana) [reply Chatify/chat/templates/chat/register.html:44]: yes without that it shows error.

File: Chatify/chat/consumers.py (hunk 9 above)
Comment (Dhruvi-Makwana) [reply Chatify/chat/consumers.py]: sir it's just function name harsh_kalal change according to requirement

**Pull request closed without merging**